ckanext/qgov/common/user_creation/validators.py

```diff
@@ -0,0 +1,32 @@
+from ckan.plugins import toolkit
+
+
+def _get_user():
+    """ Retrieve the current user object.
+    """
+    # 'g' is not a regular data structure so we can't use 'hasattr'
+    if 'userobj' in dir(toolkit.g):
+        return toolkit.g.userobj
+    else:
+        return None
+
+
+def data_qld_user_name_validator(key, data, errors, context):
+    user = _get_user()
+    is_sysadmin = user is not None and user.sysadmin
+
+    if not is_sysadmin and 'publisher' in data[key].lower():
```

Comment:
are we going to fix salsa's mess. ie. check context and see if they are indeed altering their username/display name?

Reply:
It would be a good thing to add, yes.

Reply:
This is done.